src/box_utils.cpp

```diff
@@ -74,43 +74,74 @@ Box makeBoxFromTwoPoints(const geometry_msgs::Point32& point0, const geometry_ms
   return box;
 }
 
+std::vector<Box> makeBoxArrayFromXMLRPC(const XmlRpc::XmlRpcValue& box_array_xmlrpc, const std::string& full_param_name)
+{
+  // checks if the array has at least one elements.
+  if (box_array_xmlrpc.getType() != XmlRpc::XmlRpcValue::TypeArray || box_array_xmlrpc.size() == 0)
+  {
+    ROS_FATAL("The box array (parameter %s) must have at least one box, "
+              "e.g. [[[x1, y1, z1], [x2, y2, z2]]]",
+              full_param_name.c_str());
+
+    throw std::runtime_error("The box array must have at least one box, "
+                             "e.g. [[[x1, y1, z1], [x2, y2, z2]]]");
+  }
+
+  std::vector<Box> box_array(box_array_xmlrpc.size());
+
+  for (int i = 0; i < box_array_xmlrpc.size(); ++i)
+  {
+    XmlRpc::XmlRpcValue box_xmlrpc = box_array_xmlrpc[i];
+    box_array[i] = makeBoxFromXMLRPC(box_xmlrpc, full_param_name);
+  }
```

Comment:
I think it's natural to use `std::transform` or something in such case in c++.

https://cpprefjp.github.io/reference/algorithm/transform.html

Like below (not work actually)
```suggestion
  std::vector<Box> box_array;
  box_array.reserve(box_array_xmlrpc.size());
  std::transform(box_array_xmlrpc.cbegin(), box_array_xmlrpc.cend(), [](const auto &box_rpc) { return makeBoxFromXMLRPC(box_xmlrpc, full_param_name); });
```

Reply:
Teh idea you proposed looks so hard for me. It is because the `iterator` of `XmlRpc::XmlRpcValue` is complicated, i.e. using the iterator we get not a `XmlRpc::XmlRpcValue` but a `std::pair<const std::__cxx11::basic_string<char>, XmlRpc::XmlRpcValue>’ to ‘XmlRpc::XmlRpcValue`.
http://docs.ros.org/en/jade/api/xmlrpcpp/html/classXmlRpc_1_1XmlRpcValue.html